R/correspondenceanalysis.R

```diff
@@ -238,32 +233,25 @@ CorrespondenceAnalysis = function(x,
         if (output == "Bubble Chart")
         {
             table.maindim <- ifelse(transpose, "columns", "rows")
-            if(is.null(bubble.size))
+            if (is.null(bubble.size))
                 stop("Bubble Charts require bubble sizes.")
-            bubble.size <- matchTableNames(bubble.size, rownames(x), table.maindim)
-            #bubble.names <- rownames(bubble.size)
-         #   if (is.null(bubble.names) && !is.null(names(bubble.size)))
-         #       bubble.names <- names(bubble.size)
-         #   if (is.null(bubble.names))
-         #       stop("The bubble sizes need to be named.")
-         #   bubble.size <- as.numeric(unlist(bubble.size))
-         #   if (length(table.names <- rownames(x)) != length(bubble.names))
-         #       stop("The number of bubble sizes does not match the number of ", table.maindim, " in the table.")
-         #   if (length(unique(bubble.names)) !=  length(bubble.names))
-         #       stop("There are duplicate bubble size names.")
-         #   if (!all(sort(bubble.names) == sort(table.names)))
-         #       if (!all(sort(tolower(bubble.names)) == sort(tolower(table.names))))
-         #       {
-         #           stop("The bubble sizes must contain the same names as in the ",
-         #                table.maindim, " of the input data table: ",
-         #                paste0(paste0(table.names, ":", bubble.names), collapse = ", "), ".")
-         #       }
-            # Sorting bubble sizes to match the row names of the table.
-            #order = match(rownames(x), bubble.names)
-            #if (sum(order, na.rm = TRUE) != sum(1:length(order)))
-            #    stop("The bubble sizes must contain the same names as in the rows of the input data table: ",
-            #         paste0(paste0(table.names, ":", bubble.names), collapse = ", "), ".")
-            #bubble.size = bubble.size[order]
+            bubble.size <- as.matrix(bubble.size)
+            if (is.null(rownames(bubble.size)))
+                stop("The bubble sizes need to be named to match the row labels of the input data.")
+            if (NCOL(bubble.size) > 1)
+            {
+                warning("Only first column of bubble sizes is used.")
```

Comment:
Reword this to mention: "The table of bubble sizes contains more than one column. Only first column of bubble sizes was used."

Reply:
Fixed in 35da5f2